src/backend/impl/src/controllers/proposal_review_controller.rs

```diff
@@ -135,6 +147,20 @@ impl<A: AccessControlService, P: ProposalReviewService> ProposalReviewController
         self.proposal_review_service
             .get_proposal_review(calling_principal, request)
     }
+
+    fn get_my_proposal_review(
+        &self,
+        calling_principal: Principal,
+        request: GetMyProposalReviewRequest,
+    ) -> Result<GetMyProposalReviewResponse, ApiError> {
+        self.access_control_service
+            .assert_principal_not_anonymous(&calling_principal)?;
```

Comment:
Is this assertion used just to have a more fine-grained error depending on the caller? Because the next one implicitly already includes this one

Reply:
Having the anonymous check first reduces the cycles that we will spend before we reject the call, although in this case since it's a query call it doesn't actually matter. I think it would be good to do that everywhere, but maybe it's better to include it in the access control service instead so we get it everywhere by default.

Reply:
> I think it would be good to do that everywhere, but maybe it's better to include it in the access control service instead so we get it everywhere by default.

Agree! I think we need a separate PR though, because it can break other tests